benchmarks/encoder/util.py

```diff
@@ -0,0 +1,45 @@
+import random
+import string
+
+from ddtrace.span import Span
+
+
+def rands(size=6, chars=string.ascii_uppercase + string.digits):
+    return "".join(random.choice(chars) for _ in range(size))
+
+
+# set of randomly generated span attributes to be used in traces
+SPAN_NAMES = [rands(size=16) for _ in range(256)]
+RESOURCES = [rands(size=16) for _ in range(256)]
+SERVICES = [rands(size=16) for _ in range(16)]
+TAG_KEYS = [rands(size=16) for _ in range(256)]
+METRIC_KEYS = [rands(size=16) for _ in range(256)]
+
+
+def gen_traces(ntraces=1, nspans=1, ntags=0, ltags=0, nmetrics=0):
```

Comment:
I worry a bit about how this doesn't match how we typically construct traces with the API. One example is that the service name will be inherited down the trace which doesn't quite match the logic here (although in that particular example it basically matches because each span gets a service but you get my point hopefully)

Any reason we don't use `tracer.trace()` to make these spans? Using the DummyWriter leaves the traces untouched I think which might be closer to reality.

Reply:
I think using `tracer.trace()` here would be fine, with of course disabling writing.

We don't time this method in this scenario so whatever is the easiest and most accurate data, we should go with it. 

Your point about not setting the service on each span is not something I had considered. We could change the randomization of the service names to skip setting for a certain portion of spans. We'd expect to see a drop with one less string attribute to encode. However I think this data is helpful to keep as it gives us more a worst case of the average case, if that makes sense. We repeat the value of these attributes across traces, which is closer to reality.

Reply:
:+1: